src/main/java/service/SupplyOperationHandlerImpl.java

```diff
@@ -0,0 +1,13 @@
+package service;
+
+import dao.FruitsDao;
+import dao.FruitsDaoImpl;
+
+public class SupplyOperationHandlerImpl implements OperationHandler {
+
+    @Override
+    public void doOperation(String key, String value) {
+        FruitsDao fruitsDao = new FruitsDaoImpl();
+        fruitsDao.addProduct(key, Integer.parseInt(value) + fruitsDao.getValue(key));
+    }
+}

```

Comment:
We shoudn't need to create `FruitDaoImpl` object on every `doOperation(...)`.
Lets create `FruitsDao` field inside class and initialize it by constructor
P.S. do same in every OperationHanler implementation